src/prefect/flows.py

```diff
@@ -778,6 +796,160 @@ async def from_source(
 
         return flow
 
+    @sync_compatible
+    async def deploy(
+        self,
+        name: str,
+        work_pool_name: str,
+        image: Union[str, DeploymentImage],
+        skip_push: bool = False,
```

Comment:
Any thoughts on why `skip_push=True` is preferable to `push=False`?

Reply:
Ooh, I like `push=False` better. Boolean values defaulting to false are common in Vue and React development and I think I was channeling that.

Reply:
Added in [5dd3a6a](https://github.com/PrefectHQ/prefect/pull/10975/commits/5dd3a6a7b8497cc046bb56ab481133480ef33b95)!